src/Speck.js

```diff
@@ -134,13 +134,32 @@ class Speck {
     return JSON.parse(JSON.stringify(rawData));
   }
 
+  toJSONContext(context) {
```

Comment:
We don't even have the concept of a context living on its on having an Entity on top of it, should we create this concept ? It changes the API completely. 
If we keep the context with something part of the Entity it self, the api should direct access the context, like ``` user.toJSONByContext('astronaut')``` or ``` user.toJSONContext('astronaut')```.
And in this case, I prefer  *toJSONByContext* = "Transform this entity in a JSON based on 
the fields declared on the *x* context".
Instead of 
And in this case, I prefer  *toJSONContext* which looks like : "Trans form this context in a JSON" which doesn't make any sense.

I don`t see this problem on the validate context API.